src/spatio_temporal_voxel_layer.cpp

```diff
@@ -582,35 +582,24 @@ void SpatioTemporalVoxelLayer::DynamicReconfigureCallback( \
 /*****************************************************************************/
 {
   boost::recursive_mutex::scoped_lock lock(_voxel_grid_lock);
-  bool update_grid(false);
-  auto updateFlagIfChanged = [&update_grid](auto& own, const auto& reference)
-  {
-    if (static_cast<float>(std::abs(own - reference)) >= FLT_EPSILON)
-    {
-      own = reference;
-      update_grid = true;
-    }
-  };
-
-  auto default_value = (config.track_unknown_space) ? \
-                          costmap_2d::NO_INFORMATION : costmap_2d::FREE_SPACE;
-  updateFlagIfChanged(default_value_, default_value);
-  updateFlagIfChanged(_voxel_size, config.voxel_size);
-  updateFlagIfChanged(_voxel_decay, config.voxel_decay);
-  int decay_model_int = (int)_decay_model;
-  updateFlagIfChanged(decay_model_int, config.decay_model);
-  updateFlagIfChanged(_publish_voxels, config.publish_voxel_map);
 
   _enabled = config.enabled;
   _combination_method = config.combination_method;
   _mark_threshold = config.mark_threshold;
   _update_footprint_enabled = config.update_footprint_enabled;
   _mapping_mode = config.mapping_mode;
   _map_save_duration = ros::Duration(config.map_save_duration);
-  _decay_model = static_cast<volume_grid::GlobalDecayModel>(decay_model_int);
 
-  if (update_grid)
+  if (level >=1) //update grid
```

Comment:
So, I've never used the level parameter before, so please educate me on why this would tell you if you need to update just those parameters and make a new voxel grid? The rationale behind that function was so that if things changed that requires re-instantiating the underlying grid we create a new one only when those parameters change. I think the level indicator wouldn't be the discriminator here.

Reply:
Ah, I think I understand, the level will change if 0->1 or 1->0's from the defaults. 

If that's true, what happens when you change something you changed once back down to the original? I dont think that would re-instantiate the grid object because the level would look like its opposite case - for instance as you would while parameter tuning.

Reply:
For what I understand each time reconfigure is called level parameter changes according which group of params affects. 
- If only fields with level  0 change level will be 0.
- If only fields with 1 value change level will be 1.
- If both levels field are updated level will be 0 + 1 = 1 .

Level works as a bitmask to keep track of which group of parameters has changed.
If when reconfigure callback is called level is 0 we know that all grid parameters haven't changed, so we don't update the grid. If level is 1 we update the grid.

Reply:
Got it, my bad, I thought this was defaults from my quick scanning. This makes sense